File: app/src/main/java/com/adammcneilly/nhacks/BrandActivity.java
package com.adammcneilly.nhacks;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class BrandActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_brand);

        // Get brands
        List<Brand> brandList = getSampleBrands();

        // Create adapter
        BrandAdapter adapter = new BrandAdapter(brandList);

        // Layout Manager
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);

        // Setup RecyclerView
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.brand_list);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setAdapter(adapter);
    }

    /**
     * @return A collection of Brands to display in the application.
     */
    private List<Brand> getSampleBrands() {
        List<Brand> brandList = new ArrayList<>();
        brandList.add(new Brand("Apple", R.drawable.apple));
        brandList.add(new Brand("Facebook", R.drawable.facebook));
        brandList.add(new Brand("Google", R.drawable.google));
        brandList.add(new Brand("Microsoft", R.drawable.microsoft));
        return brandList;
    }
}

File: app-kotlin/src/main/java/com/adammcneilly/demo/BrandActivity.kt
package com.adammcneilly.demo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

/**
 * Activity used to display a list of brands to the user.
 */
class BrandActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_brand)

        // Get brands
        val brandList = sampleBrands

        // Create adapter
        val adapter = BrandAdapter(brandList)

        // Layout Manager
        val linearLayoutManager = LinearLayoutManager(this)

        // Setup RecyclerView
        val recyclerView = findViewById(R.id.brand_list) as RecyclerView
        recyclerView.layoutManager = linearLayoutManager
        recyclerView.adapter = adapter
    }

    /**
     * List of hardcoded brands to display to the user.
     */
    private val sampleBrands = listOf(Brand("Apple", R.drawable.apple), Brand("Facebook", R.drawable.facebook), Brand("Google", R.drawable.google), Brand("Microsoft", R.drawable.microsoft))
}